### ISSUE TITLE
out of memory

### ISSUE THREAD

**Issue by marudor** [webpack/webpack#2157]
Webpack 2 terminates due to out of memory for our Project at work.

```
<--- Last few GCs --->

  117575 ms: Mark-sweep 1356.4 (1457.2) -> 1364.8 (1457.2) MB, 593.4 / 0 ms [allocation failure] [scavenge might not succeed].
  118168 ms: Mark-sweep 1364.8 (1457.2) -> 1364.8 (1457.2) MB, 592.5 / 0 ms [allocation failure] [scavenge might not succeed].
  118777 ms: Mark-sweep 1364.8 (1457.2) -> 1364.8 (1457.2) MB, 609.1 / 0 ms [last resort gc].
  119381 ms: Mark-sweep 1364.8 (1457.2) -> 1364.8 (1457.2) MB, 603.8 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x2caa627e3ac1 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [/Users/entwickler/programming/factroFrontend/node_modules/webpack/lib/optimize/RemoveParentModulesPlugin.js:43] [pc=0xe09107f767b] (this=0x2caa62704189 <undefined>,key=0x11a5e8dd09d9 <String[4]: 1084>)
    2: arguments adaptor frame: 3->1
    3: InnerArrayForEach(aka InnerArrayForEach) [native array.js:~942] [pc=0xe09109b31f0] (this=0x2caa62704189 <u...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
fish: 'node_modules/.bin/webpack' terminated by signal SIGABRT (Abort)
```

Sadly I'm not allowed to just give you our Project for debugging.
I hope you can still do something about it.

Configuration:

``` js
eslint: {
    configFile: './.eslintrc',
    failOnWarning: false,
    failOnError: true,
  },
  context: __dirname,
  resolve: {
    modulesDirectories: ['node_modules/@marudor', 'dependency', 'node_modules'],
    alias: {
      theme: 'Theme/index.js',
      warning: 'fbjs/lib/warning.js',
      shallowequal: 'fbjs/lib/shallowEqual.js',
      vex: 'vex/js/vex.js',
      joyride: 'react-joyride/index.js',
      vexCSS: path.resolve('./dependency/vex/css'),
    },
  },
  entry,
  output: {
    path: outputPath,
    filename: name,
    chunkFilename: node_env === 'production' ? '[id]-[chunkhash].js' : '[id].js',
    publicPath: '/',
  },
  module: {
    loaders: [
      { test: /^((?!(CSS|css)\.js$).)*(\.jsx?)$/,
        exclude: /(node_modules|dependency|primusClient\.js|Unit\/Services)/,
        loader: jsLoader,
      },
      {
        test: /tinymce(\\|\/)(plugins|themes|skins|langs)/,
        loader: 'file?name=[path][name].[ext]&context=dependency/tinymce',
      },
      { test: /\.css$/, loader: ExtractTextPlugin.extract('style', 'css'), include: /Editor/, exclude: /tinymce(\\|\/)/ },
      { test: /\.css$/, loader: 'style!css', exclude: /(tinymce(\\|\/)|Editor)/ },
      { test: /\.(CSS|css)\.js$/, exclude: /(node_modules|dependency)/, loader: `inline-css!${jsLoader}` },
      { test: /\.(jpg|png|gif|jpeg|ico)$/, loader: 'url', exclude: /tinymce(\\|\/)/ },
      { test: /\.(eot|ttf|otf|svg|woff2?)(\?.*)?$/, loader: 'file', exclude: /tinymce(\\|\/)/ },
      { test: /\.pdf$/, loader: 'file', exclude: /tinymce(\\|\/)/ },
      { test: /\.json$/, loader: 'json' },
      {
        test: /-spec\.js/,
        exclude: /(node_modules|dependency)/,
        loader: 'mocha!babel!eslint',
      },
    ],
    noParse: [
      /primusClient\.js/,
      /.*primusClient.*/,
      /react\\dist\\react(-with-addons)?\.js/,
    ],
  },
  plugins: [
    new webpack.NoErrorsPlugin(),
    new webpack.DefinePlugin({
      'process.env': {
        NODE_ENV: JSON.stringify(node_env),
      },
      FACTRO_CONFIG: JSON.stringify(config),
      __DEV__: JSON.stringify(__DEV__),
      __PROD__: JSON.stringify(!__DEV__),
      __WEB__: JSON.stringify(true),
      __APP__: JSON.stringify(false),
    }),
    new ExtractTextPlugin('[name].css'),
    new HtmlWebpackPlugin({
      filename: 'index.html',
      template: 'html!src/Web/index.html',
      minify: {},
    }),
    new webpack.optimize.CommonsChunkPlugin({
      async: true,
      minChunks: 3,
      children: true,
    }),
  ]
  devServer: {
    proxy: {
      '/api/*': {
        target: 'http://localhost:65429',
        toProxy: true,
        changeOrigin: true,
        xfwd: true,
      },
      '/primus/*': {
        target: 'http://127.0.0.1:3000',
        ws: true,
      },
    },
  },
```

**Comment by sokra** [comment]
Is it a big project? If so, you can increase the memory available to the node.js process...

**Comment by marudor** [comment]
Well - I did it up to 10 GB.
Takes about 10 minutes - then crashes as well.
Did not try to boost it even further as I think if it wont run with 10GB it just wont run...
(iMac 21", 16GB Ram, 3,3ghz i7)

**Comment by sokra** [comment]
try to downgrade to 2.1.0-beta.3 until we can fix it...

**Comment by TylerYang** [comment]
Hi folks , same issue here.

```

<--- Last few GCs --->

24586627 ms: Scavenge 1393.6 (1452.5) -> 1393.6 (1452.5) MB, 0.9 / 0 ms (+ 2.1 ms in 1 steps since last GC) [allocation failure] [incremental marking delaying mark-sweep].
24587305 ms: Mark-sweep 1393.6 (1452.5) -> 1393.6 (1452.5) MB, 678.8 / 0 ms (+ 3.6 ms in 2 steps since start of marking, biggest step 2.1 ms) [last resort gc].
24587931 ms: Mark-sweep 1393.6 (1452.5) -> 1393.6 (1452.5) MB, 625.7 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3d5f94644a49 <JS Object>
    1: /* anonymous */(aka /* anonymous */) [/Users/tyler/Work/whatever/node_modules/webpack-core/node_modules/source-map/lib/source-map/source-node.js:~341] [pc=0x27a2aa7bbe95] (this=0x3d5f94604139 <undefined>,chunk=0x243a07470781 <String[55]\: \x09        var thisBinding = baseCreate(Ctor.prototype),\n>,original=0x33954213bcd9 <an Object with map 0x267a24b58fd9>)...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
Abort trap: 6
```
"babel": "^6.5.2",
    "babel-core": "^6.7.4",
    "babel-eslint": "^5.0.0",
    "babel-loader": "^6.2.4",
    "babel-plugin-transform-object-assign": "^6.5.0",
    "babel-plugin-transform-react-jsx": "^6.7.4",
    "babel-plugin-transform-runtime": "^6.6.0",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "babel-runtime": "^6.6.1",
    "console-stamp": "^0.2.0",
    "css-loader": "^0.23.1",
    "eslint": "^2.3.0",
    "eslint-config-airbnb": "^6.1.0",
    "eslint-loader": "^1.3.0",
    "eslint-plugin-flow-vars": "^0.2.1",
    "eslint-plugin-react": "^4.2.1",
    "estraverse-fb": "^1.3.1",
    "html-webpack-plugin": "^2.10.0",
    "image-webpack-loader": "^1.6.3",
    "json-loader": "^0.5.4",
    "less": "^2.6.1",
    "less-loader": "^2.2.3",
    "style-loader": "^0.13.0",
    "url-loader": "^0.5.7",
    "webpack": "^1.12.14",
    "webpack-dev-server": "^1.14.1"
```
My webpack version is **1.12.14**, devDependencies,



@marudor Did @sokra 's solution solve this problem?

**Comment by marudor** [comment]
@TylerYang nope,
Even with older Webpack2 versions it broke.
You're using webpack1 though. Works for me with that.

**Comment by TylerYang** [comment]
@marudor  Thx, do you mean it works(won't crash anymore) when you switch back to webpack1? 

Actually I did meet this issue several months ago.(With webpack1, also) and I was thinking something is wrong with my environment. (Maybe 3rd party dependencies)  

Seems for now I have to restart the devServer after a few hours.  Would really like to dig into that to see what could possibly go wrong.

**Comment by achicu** [comment]
In our project a V8 issue was causing node to run out of memory. I've sent a PR to workaround the issue in webpack: https://github.com/webpack/webpack/pull/2497

**Comment by TheLarkInn** [comment]
@TylerYang @marudor 

Does the V8 memory issues that @achicu described look related to your issues. Please list the following if this is still an issue: 

* Node version
  * If possible, can you update your version of node, and does updating fix the problem
* Webpack 2 version: 
  * If possible, update to most recent webpack beta, does this fix your problem

**Comment by marudor** [comment]
@TheLarkInn Node 6.2.2, webpack just updated.
Still OOM.
With Node 6.0-6.1 I also get crashes with webpack 1.

**Comment by sokra** [comment]
Did you try to increase the node.js memory? `--max_old_space_size=<size>` in MB

**Comment by szagi3891** [comment]
@skora: I tried. Also, there is a problem.

**Comment by rochdev** [comment]
Same issue for a very small project

**Comment by KhoaSydney** [comment]
Hi,
I'm using webpack 2.2.0. It started the build and took a while then fail with error process out of memory. Any updates on this issue?

Thanks,
Khoa

**Comment by TheLarkInn** [comment]
Can anyone reproduce this In a tiny example? I'd like to pull it down and investigate further.

**Comment by tlenex** [comment]
Same issue here, happens often on 8GB RAM machine

**Comment by pkantsedalov** [comment]
The same as in [tlenex's](https://github.com/webpack/webpack/issues/2157#issuecomment-279246738) case

**Comment by gribnoysup** [comment]
Not sure that this is the same issue that is discussed here, but maybe this is somehow related.

I tried to upgrade webpack from 2.2.0 to 2.2.1 recently and build started crashing with this:

```sh
<--- Last few GCs --->

  183621 ms: Mark-sweep 1137.3 (1447.2) -> 1135.8 (1447.2) MB, 170.5 / 0.0 ms [allocation failure] [GC in old space requested].
  183755 ms: Mark-sweep 1135.8 (1447.2) -> 1135.8 (1447.2) MB, 133.5 / 0.0 ms [allocation failure] [GC in old space requested].
  183893 ms: Mark-sweep 1135.8 (1447.2) -> 1143.6 (1428.2) MB, 138.6 / 0.0 ms [last resort gc].
  184033 ms: Mark-sweep 1143.6 (1428.2) -> 1152.5 (1427.2) MB, 139.9 / 0.0 ms [last resort gc].

<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x18f82becfb51 <JS Object>
    1: [Symbol.replace] [native regexp.js:~540] [pc=0x3d862722a3dc] (this=0x2c8e64784da1 <JS RegExp>,D=0x23bd22004201 <Very long string[370669]>,aw=0x3356ec2276c9 <String[1]: />)
    2: 0x18f82be8aaf1 <Symbol: 20329980 <String[14]: Symbol.replace>>(aka [Symbol.replace]) [native regexp.js:~659] [pc=0x3d8626dbda4b] (this=0x2c8e64784da1 <JS RegExp>,D=0x23bd22004201 <Very long string[370669]>,aw=0x...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x109608c [node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [node]
 6: 0xee705c [node]
 7: 0xeea651 [node]
 8: v8::internal::Runtime_StringReplaceGlobalRegExpWithString(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 9: 0x3d86202092a7
Aborted (core dumped)
```

I was able to narrow down this problem to `stylus-loader` and then I found this issue in `stylus-loader` repo: https://github.com/shama/stylus-loader/issues/162#issuecomment-277722106

I'm not very familiar with how loaders internals work, so it is quite difficult to me to identify what causing this in `stylus-loader` after update to `webpack@2.2.1`, but I can totally see A LOT of paths added to some kind of paths cache somewhere in loader internals (there is a link to src in issue comment above), and this is not happening on `2.2.0`.

@sokra @TheLarkInn sorry for pinging you guys like that, but maybe you (or someone else) could suggest what changes in 2.2.1 compared to 2.2.0 could cause that?

**Comment by sresant** [comment]
I am also seeing this issue. I noticed that if i were to use say the `cheap-source-map` instead or any other devtool it works fine but this problems seems specific to `source-map` devtool for my case.

**Comment by m-andrew-albright** [comment]
I too am experiencing this issue.  Node 7.5.0, Windows 10 with 16GB RAM, Powershell.

**Comment by sresant** [comment]
Update: My issue ended up being in the extract-text plugin(ETP)....To start the sourcemap dev tool plugin is a memory hog but the real issue was that ETP was generating some huge css files due to https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/452 which increased the memory usage of the sourcemap plugin. Applying that patch to ETP fixed my issue.

**Comment by terencechow** [comment]
Also experiencing this issue. App is at 550mb memory on initial start. Would like to lower this significantly since we develop in a VM with other microservices. Can we combine a list of things we can do to lower the memory footprint while developing? My VM keeps sending out of memory sigkills to this node process because it's high memory usage.

**Comment by sokra** [comment]
You should not compile on the production webserver. Best compile on CI server or dev compiler and deploy compiled assets.

**Comment by terencechow** [comment]
We aren't compiling on a CI server. We do our development in a minikube through virtual box. It allows us to mimick our production microservices architecture by running a handful of services.

**Comment by tomasztomys** [comment]
The same situation in my project.

**Comment by liadoster** [comment]
Same issue in my project.
it's happened only when devtool set to "source-map" (when devtool set to "cheap-source-map" it work fine).

**Comment by about-code** [comment]
Given reports which indicate that the issue might relate to *source maps* I had a quick look at `SourceMapDevToolPlugin.js`. In [line 60](https://github.com/webpack/webpack/blob/215647d9fcf2cfa69c83b3721ae7058ccd31d851/lib/SourceMapDevToolPlugin.js#L60) array entries are pushed onto `chunk.files` within a `map()`-callback used when iterating the same array. Its an uneducated guess but modifying an array while iterating over it is always prone to side effects. I could imagine `chunk.files` is stacking up over time, though can't verify it at the moment. But since we push cached files onto the array it might fit into reports telling that the problem occurs only after a few rebuilds.

**Comment by about-code** [comment]
Further the `filter().map().forEach()` chain over `chunk.files` maps to `tasks` which when iterated later will further add to `chunk.files` in [line 158](https://github.com/webpack/webpack/blob/215647d9fcf2cfa69c83b3721ae7058ccd31d851/lib/SourceMapDevToolPlugin.js#L158). 

These are only results from a superficial review for code smells made with a worst-case assumption. I've not attempted to fully understand the code (would have taken hours or days, probably). So read this as a disclaimer, please.

**Comment by MagicDuck** [comment]
Has anybody tried to run the dev tools memory profiler on an incremental build? That's how i found this problem with the source-map library a while ago: https://github.com/mozilla/source-map/issues/259

Don't want to seem too negative, but source mapping is a critical part of webpack and the source-map lib from mozilla has piss poor performance. It was written in an era of tiny javascript.
I think the ideal would be a native implementation. I remember seeing an incomplete rust one which was 100x faster...

**Comment by about-code** [comment]
For the record: *ModuleConcatenationPlugin* seems to have been another cause for out-of-memory issues but fixed https://github.com/webpack/webpack/issues/5992

**Comment by hultberg** [comment]
I'm just leaving this error dump here. I'm using webpack-dev-server and after doing some work and emitting several changes with hot-module-reload the server crashes by using too much memory. I have been watching the memory usage and noticed it always grows after an emit and will never decrease (memory leak?)

Project is about 9MB of builds (without source maps)

```
<--- Last few GCs --->

[16:0x55a8aa6e0000] 18017361 ms: Mark-sweep 1358.1 (1415.2) -> 1358.0 (1415.2) MB, 160.8 / 0.1 ms  allocation failure GC in old space requested
[16:0x55a8aa6e0000] 18017524 ms: Mark-sweep 1358.0 (1415.2) -> 1357.9 (1378.7) MB, 162.6 / 0.1 ms  last resort GC in old space requested
[16:0x55a8aa6e0000] 18017699 ms: Mark-sweep 1357.9 (1378.7) -> 1357.9 (1378.7) MB, 175.6 / 0.1 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1ffc11a5749 <JSObject>
    1: fromString(aka fromString) [buffer.js:340] [bytecode=0xfd9f52437d1 offset=161](this=0x13970f1822d1 <undefined>,string=0x2b66c3dddb49 <Very long string[25606263]>,encoding=0x1ffc11b49c1 <String[4]: utf8>)
    2: from [buffer.js:193] [bytecode=0xfd9f52432b1 offset=14](this=0xa4009d3b819 <JSFunction Buffer (sfi = 0x1ffc11fdd19)>,value=0x2b66c3dddb49 <Very long string[25606263]>,encodingOrOffs...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```

**Comment by about-code** [comment]
@hultberg The `<Very long string[25606263]>` message makes me feel it could result from the ModuleConcatenationPlugin bug in #5992 which has been reported to create very long module identifier strings leading to OOMs.

**Comment by hultberg** [comment]
@about-code Thanks, but I don't have that plugin enabled in my project (is it a default?), but I'm suspecting [webpack-manifest-plugin](https://github.com/danethurber/webpack-manifest-plugin) as I have it enabled during hot-reload to have an overview over assets created. When I disable it, the memory usage gets high, but it seems the GC is working as memory is cleaned, preventing crashes.

**Comment by MagicDuck** [comment]
@hultberg can you start with node --inspect-brk and then run a memory alloction profile of your build? I remember seeing Compiler writeOut() function allocating a lot in my case when it converts strings to buffers for writing (which i find crazy). We fixed it for our build by modifying our in house indexed sourcemap plugin to generate raw sources containing buffers instead of strings.

**Comment by Anthony59273** [comment]
I'm also having about the same error directly when running WDS. Here are the details:

- Operating System: MacOS 10.12.6
- Node Version: 4.4.3
- NPM Version: 2.15.1
- webpack Version: 3.8.1
- webpack-dev-server Version: 2.9.2

# Code

webpack.config.js
https://gist.github.com/Anthony59273/7329a0a43d972978d09b8f1cdc944173

# Expected Behavior

Dev server runs correctly.

# Actual Behavior

When launching WDS I have the following behavior:

```
> cross-env NODE_ENV=development webpack-dev-server --hot --progress

 10% building modules 3/3 modules 0 active                                         
Project is running at http://localhost:8000/
webpack output is served from /intelligence/
 91% additional asset processing                                                          
<--- Last few GCs --->

  311753 ms: Scavenge 1376.0 (1453.5) -> 1376.0 (1453.5) MB, 0.8 / 0 ms (+ 3.2 ms in 1 steps since last GC) [allocation failure] [incremental marking delaying mark-sweep].
  312567 ms: Mark-sweep 1376.0 (1453.5) -> 1376.0 (1453.5) MB, 813.9 / 0 ms (+ 5.0 ms in 2 steps since start of marking, biggest step 3.2 ms) [last resort gc].
  313397 ms: Mark-sweep 1376.0 (1453.5) -> 1376.0 (1453.5) MB, 830.8 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1e5e919b4629 <JS Object>
    2: /* anonymous */(aka /* anonymous */) [0x1e5e919041b9 <undefined>:~1836] [pc=0x34a2da1308e7] (this=0x1e5e919041b9 <undefined>,quote_char=0x1e5e919bdaf9 <String[1]: ">)
    3: /* anonymous */(aka /* anonymous */) [0x1e5e919041b9 <undefined>:1974] [pc=0x34a2da16ca89] (this=0x1e5e919041b9 <undefined>,x=0x1e5e919bdaf9 <String[1]: ">)
    4: input(aka next_token) [0x1e5e919041b9 <undefined>:~1...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```
So here I'm having a memory issue but I actually don't know where this comes from. I tried to allow more memory to this process using node --max_old_space_size=4096 but it didn't help. This happens directly when I'm trying to launch the dev server, while before I used to experience this after a few hot reloads.

Any help would be greatly appreciated. Thanks !

**Comment by artivilla** [comment]
@Anthony59273 Try a larger size might help? I'm using `"webpack:dev": "NODE_ENV=development node --max_old_space_size=8096 ./node_modules/.bin/webpack --progress --colors",`

**Comment by Anthony59273** [comment]
@artivilla I've tried with `node --max_old_space_size=4096` without success. My WDS process takes about 1.5GB of memory usage so this value (4096) should be enough but I had no result with it.

**Comment by jsankarlal** [comment]
i am facing the same issue here.

**Comment by MagicDuck** [comment]
Friendly advice: giving version numbers is usually not enough info to get help. You have to think about it from the perspective of the project maintainer. I know i was making this mistake before and wondering why no help was fothcoming 😁. Either provide a repro repository or if that's not possible, try to get some kind of performance log/dump such as an allocation snapshot. It's actually quite easy to do. Ex insructions here: https://www.toptal.com/nodejs/debugging-memory-leaks-node-js-applications

**Comment by TheLarkInn** [comment]
Yes second this!! We in fact won't even look into this without a snapshot, heapdump or memtracing profile!

**Comment by webdevnerdstuff** [comment]
Updated to Webpack 4, still have the stacktrace crashing problem. 

I have also tried the `node --max_old_space_size=4096` solution, but that does not fix the problem. My work computer has 32GB of RAM, so memory is not an issue. But on systems with not as much RAM it does crash more often. 

I'm not using ModuleConcatenationPlugin either. 

I'm at a complete loss and our entire team has this problem. Some days it crashes the dev server upwards of 50+ times. Frustrating to say the least. 

**Latest Error:**
<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x370ca3da5e91 <JSObject>
    2: fromString(aka fromString) [buffer.js:312] [bytecode=0x370c40c7ed79 offset=164](this=0x370cf7502311 <undefined>,string=0x370c4cfd48c9 <Very long string[4772628]>,encoding=0x370ca3db6569 <String[4]: utf8>)
    4: from [buffer.js:177] [bytecode=0x370c40c7e879 offset=11](this=0x370c40c30749 <JSFunction Buffer (sfi = 0x370ca3df0781)>,value=0x370c4cfd48c9 <Very long string[4772628]>,encodingOr...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/usr/local/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/usr/local/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/usr/local/bin/node]
 4: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/usr/local/bin/node]
 5: v8::internal::String::SlowFlatten(v8::internal::Handle<v8::internal::ConsString>, v8::internal::PretenureFlag) [/usr/local/bin/node]
 6: v8::String::WriteUtf8(char*, int, int*, int) const [/usr/local/bin/node]
 7: node::StringBytes::Write(v8::Isolate*, char*, unsigned long, v8::Local<v8::Value>, node::encoding, int*) [/usr/local/bin/node]
 8: node::Buffer::New(v8::Isolate*, v8::Local<v8::String>, node::encoding) [/usr/local/bin/node]
 9: node::Buffer::(anonymous namespace)::CreateFromString(v8::FunctionCallbackInfo<v8::Value> const&) [/usr/local/bin/node]
10: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [/usr/local/bin/node]
11: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/usr/local/bin/node]
12: v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/usr/local/bin/node]
13: 0x67b7658463d

========================================================
**Here is my package dependencies for the versions:**

"dependencies": {
    "axios": "0.16.2",
    "bootstrap": "3.3.7",
    "circular-json": "0.3.1",
    "eonasdan-bootstrap-datetimepicker": "4.17.47",
    "fingerprintjs2": "1.5.1",
    "fullcalendar": "3.4.0",
    "jquery": "3.2.1",
    "json-to-form-data": "1.0.0",
    "lodash": "^4.17.5",
    "moment": "2.18.1",
    "moment-timezone": "^0.5.14",
    "object-to-formdata": "1.0.12",
    "pikaday": "1.6.1",
    "superagent": "3.5.2",
    "underscore": "1.8.3",
    "underscore.string": "^3.3.4",
    "v-tooltip": "2.0.0-beta.5",
    "velocity-animate": "1.5.0",
    "vue": "2.4.1",
    "vue-cookie": "^1.1.4",
    "vue-form-generator": "2.2.0",
    "vue-form-wizard": "0.8.1",
    "vue-image-lightbox": "5.5.3",
    "vue-instant-pdf-viewer": "1.0.2",
    "vue-js-toggle-button": "1.1.2",
    "vue-lazyload": "1.0.6",
    "vue-loader": "14.2.1",
    "vue-multiselect": "^2.0.8",
    "vue-strap": "github:wffranco/vue-strap",
    "vue-tables-2": "1.0.0",
    "vue-template-compiler": "2.4.1",
    "vue-touch": "2.0.0-beta.4",
    "vue-virtual-scroll-list": "1.0.7",
    "vue2-animate": "1.0.4",
    "vuebar": "0.0.16",
    "vuejs-datepicker": "0.9.0",
    "webpack": "4.1.1",
    "webpack-bundle-tracker": "0.2.0",
    "webpack-cli": "^2.0.12",
    "xmlhttprequest": "1.8.0"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.2.2",
    "babel-helper-vue-jsx-merge-props": "^2.0.3",
    "babel-loader": "^7.1.4",
    "babel-plugin-syntax-jsx": "^6.18.0",
    "babel-plugin-transform-vue-jsx": "^3.7.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-stage-2": "^6.24.1",
    "babel-preset-vue-app": "^2.0.0",
    "cross-env": "^5.1.4",
    "css-loader": "^0.28.11",
    "eslint": "^4.19.0",
    "eslint-config-airbnb-base": "^12.1.0",
    "eslint-friendly-formatter": "^3.0.0",
    "eslint-loader": "^2.0.0",
    "eslint-plugin-html": "^4.0.2",
    "eslint-plugin-import": "^2.9.0",
    "eslint-plugin-vue": "^4.3.0",
    "file-loader": "^1.1.11",
    "less": "^3.0.1",
    "less-plugin-clean-css": "^1.5.1",
    "node-sass": "^4.7.2",
    "sass-loader": "^6.0.7",
    "script-loader": "^0.7.2",
    "style-loader": "^0.20.3",
    "uglifyjs-webpack-plugin": "^1.2.4",
    "webpack-dev-server": "^3.1.1",
    "webpack-parallel-uglify-plugin": "^1.1.0"
  }
}

**Comment by evilebottnawi** [comment]
Thanks everyone, we have a lot of difference issues here, it is hard to track and resolve their. When you have very very very big project you really need to increase node memory (i.e. `node --max_old_space_size=4096`). Also some loaders/plugins can have memory leak.

If you faced with `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`:
1. Try to update all deps to latest version (problem can be solved).
2. Try to increase node memory - `node --max_old_space_size=4096`
3. Try to disable plugins/loaders step by step.

If nothing has helped feel free create new issue with minimum reproducible test repo. Thanks!

**Comment by akalitenya** [comment]
on node v8+

instead of 
...
  "scripts": {
    "build": "node --max_old_space_size=8192 node_modules/.bin/webpack "
  },
...

try to add this to your bash script (or .bash_profile or .bashrc):
...
export NODE_OPTIONS=--max_old_space_size=8192
...

it works for me on mac osx

[more info on stackoverflow](https://stackoverflow.com/questions/26094420/fatal-error-call-and-retry-last-allocation-failed-process-out-of-memory/48895989#48895989)

**Comment by ycjcl868** [comment]
same issue in ci.
